pkg/manager/member/tidb_member_manager.go

```diff
@@ -245,6 +257,106 @@ func (m *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.TidbC
 	return mngerutils.UpdateStatefulSetWithPrecheck(m.deps, tc, "FailedUpdateTiDBSTS", newTiDBSet, oldTiDBSet)
 }
 
+func (m *tidbMemberManager) syncInitializer(tc *v1alpha1.TidbCluster) {
+	// set random password
+	ns := tc.Namespace
+	tcName := tc.Name
+	//check endpoints ready
+	isTiDBReady := false
+	eps, epErr := m.deps.EndpointLister.Endpoints(tc.Namespace).Get(controller.TiDBMemberName(tc.Name))
+	if epErr != nil {
+		klog.Errorf("Failed to get endpoints %s for cluster %s/%s, err: %s", controller.TiDBMemberName(tc.Name), ns, tcName, epErr)
+		return
+	}
+	// TiDB service has endpoints
+	if eps != nil && len(eps.Subsets[0].Addresses) > 0 {
+		isTiDBReady = true
+	}
+
+	if !isTiDBReady {
+		klog.Infof("Wait for TiDB ready for cluster %s/%s", ns, tcName)
+		return
+	}
+	// sync password secret
+	var password string
+	secretName := controller.TiDBInitSecret(tc.Name)
+	secret, err := m.deps.SecretLister.Secrets(tc.Namespace).Get(secretName)
+	passwordSecretExist := true
+	if err != nil {
+		if errors.IsNotFound(err) {
+			passwordSecretExist = false
+		} else {
+			klog.Errorf("Failed to get endpoints %s for cluster %s/%s, err: %s", controller.TiDBMemberName(tc.Name), ns, tcName, epErr)
+			return
+		}
+	}
+
+	if !passwordSecretExist {
+		klog.Infof("Create random password for cluster[%s:%s]", tc.Namespace, tc.Name)
+		var secret *corev1.Secret
+		secret, password = m.buildRandomPasswordSecret(tc)
+		err := m.deps.TypedControl.Create(tc, secret)
+		if err != nil {
+			klog.Errorf("Failed to create secret[%s:%s], err: %s", secret.Namespace, secret.Name, err)
+			return
+		}
+	} else {
+		password = string(secret.Data[constants.TidbRootKey])
+	}
+	// init password
+	var db *sql.DB
+	var dsn string
+	err = wait.PollImmediate(1*time.Second, 5*time.Second, func() (done bool, err error) {
```

Comment:
What about we do not poll here and just wait for the next sync if any error occurs.

Reply:
ok